builder/build.go

```diff
@@ -79,7 +79,10 @@ func Build(pkgName, outpath string, config *compileopts.Config, action func(Buil
 	// keep functions interoperable, pass int64 types as pointers to
 	// stack-allocated values.
 	// Use -wasm-abi=generic to disable this behaviour.
-	if config.Options.WasmAbi == "js" && strings.HasPrefix(config.Triple(), "wasm") {
+	if config.Options.WasmAbi != "" {
+		config.Target.WasmAbi = config.Options.WasmAbi
```

Comment:
Please do not modify the `config.Target` field, it should contain exactly what is extracted from the JSON files.

Instead, you can either:

 * Add a getter to the `config` variable (`*compileopts.Config`), so you can simply call `config.WasmAbi()` to get the value.
 * Use a local variable instead.

The getter would be slightly cleaner, as it matches the pattern of other configurations (`CGO_ENABLED`, `GC`, `NeedsStackObjects`, etc).

Reply:
added the getter in https://github.com/tinygo-org/tinygo/pull/1421/commits/d61d49c69b4ce0032b527089db852d8984bcf331

Reply:
@mathetake thanks for the addition. 

@aykevl anything else needed for us to merge this now?